### DISCUSSION TITLE
Load balancer

### DISCUSSION THREAD

**Discussion by carstenjaksch** [pocketbase/pocketbase#515] [Q&A] [answered]
Hi 👋,

is it possible to serve multiple PocketBase executables with persistent storage and a load balancer?

I think of one Hetzner block storage volume with the data folder (both database and file storage), one executable per cloud server (the block storage is attached to every server) and a load balancer in front.

The only pitfall I could think of is loss of data because multiple instances could try to write the same data. 🤔

**Comment by tonyhart7** [comment]
in future maybe, Yes

you realise pocketbase ship sqlite on wal mode right, at this time pocketbase focus is in vertical scaling on single machine

there are lot of option for your needs, pocketbase isnt the one

**Comment by ganigeorgiev** [comment, marked as answer]
@carstenjaksch 

While technically it may be possible to have multiple applications connected to the same db file, you have to keep in mind that SQLite allows only a single a writter at a time (they will queue up). You can also use something like Marmot - it was recently posted here https://github.com/pocketbase/pocketbase/discussions/487.

It depends on how you plan to use PocketBase, but the embedded database is not the only thing that prevents PocketBase to scale horizontally, because there are a lot of other parts of the application that you'll have to take care for - syncing realtime events, avoiding duplicated event hook calls, etc.

Why do you want to go with this approach? A lot of developers seems "afraid" of a single server setup, but my recommendation would be to benchmark based on your use case, because often it may turn out that a single server would be more than enough.

As the other users mentioned, there also alternative platforms that you may want to check if you really need horizontal scalling - Supabase, Nhost, Appwrite, SurrealDB, etc.

**Reply by carstenjaksch** [reply]
Thanks, I was just curious.

**Comment by pchasco** [comment]
Assuming your workload is typical and most database activity is reads, you may be able to host multiple instances as long as you ensure you only ever write to one instance. This requires your application to maintain at least two PocketBase clients in your application; one for reads, and one for writes. The read-only replicas would be kept in sync by using something like Litesync. There are a lot of downsides to an approach like this, though, however these downsides are present in other platforms using a single master approach

**Reply by pchasco** [reply]
Also keep in mind this only really makes sense if you maintain multiple read replicas and balance between them.
